flexmeasures/data/models/planning/shiftable_load.py

```diff
@@ -149,6 +139,56 @@ def compute(self) -> pd.Series | None:
 
         return schedule.tz_convert(self.start.tzinfo)
 
+    def block_invalid_starting_times_for_whole_process_scheduling(
```

Comment:
This util function is great but it and its usage still needs work imo:

- It should only do what it says in the title. Checking feasibility should happen in the actual computation functions.
- Following that, it should only be called when `load_type in [LoadType.INFLEXIBLE, LoadType.SHIFTABLE]`.
- When used, the variable containing the returned value should not be called `time_restrictions`. It is now `start_time_restrictions`.